packages/store-api/src/platforms/vtex/resolvers/validateCart.ts

```diff
@@ -0,0 +1,162 @@
+import type { IStoreCart, IStoreOffer } from '../../../__generated__/schema'
+import type {
+  OrderForm,
+  OrderFormItem,
+  OrderFormInputItem,
+} from '../clients/commerce/types/OrderForm'
+import type { Context } from '..'
+
+const getId = (item: IStoreOffer) =>
+  [item.itemOffered.sku, item.seller.identifier, item.price].join('::')
+
+const orderFormItemToOffer = (
+  item: OrderFormItem,
+  index?: number
+): IStoreOffer & { index?: number } => ({
+  listPrice: item.listPrice / 100,
+  price: item.sellingPrice / 100,
+  quantity: item.quantity,
+  seller: { identifier: item.seller },
+  itemOffered: {
+    sku: item.id,
+    image: [],
+    name: item.name,
+  },
+  index,
+})
+
+const offerToOrderItemInput = (
+  offer: IStoreOffer & { index?: number }
+): OrderFormInputItem => ({
+  quantity: offer.quantity,
+  seller: offer.seller.identifier,
+  id: offer.itemOffered.sku,
+  index: offer.index,
+})
+
+const groupById = (offers: IStoreOffer[]): Map<string, IStoreOffer> =>
+  offers.reduce((acc, item) => {
+    const id = getId(item)
+
+    acc.set(id, acc.get(id) ?? item)
+
+    return acc
+  }, new Map<string, IStoreOffer>())
+
+const equals = (of1: OrderForm, of2: OrderForm) => {
+  const pick = ({ orderFormId, messages, items, salesChannel }: OrderForm) => ({
+    orderFormId,
+    messages,
+    salesChannel,
+    items: items.map(
+      ({ uniqueId, quantity, seller, sellingPrice, availability }) => ({
+        uniqueId,
+        quantity,
+        seller,
+        sellingPrice,
+        availability,
+      })
+    ),
+  })
+
+  return JSON.stringify(pick(of1)) === JSON.stringify(pick(of2))
```

Comment:
Do you think comparing a big string is a good idea, isn't it?

I have a thin algorithm that is a little bit performant:

```ts
const comparison = (object1: any, object2: any) => {
  const keys = Object.keys(object1);

  for (let i = 0; i < keys.length; i++) {
    if (
      typeof object1[keys[i]] === "string" ||
      typeof object1[keys[i]] === "number"
    ) {
      if (object1[keys[i]] !== object2[keys[i]]) {
        return false;
      }
    }

    if (
      (Array.isArray(object1[keys[i]]) ||
        typeof object1[keys[i]] === "object") &&
      !comparison(object1[keys[i]], object2[keys[i]])
    ) {
      return false;
    }
  }

  return true;
};
```

Reply:
https://codesandbox.io/s/typescript-forked-o67fh?file=/src/index.ts:0-537

Reply:
I can try to use [fast-deep-equal](https://www.npmjs.com/package/fast-deep-equal), what do you think?

Reply:
Yes, is better to use this lib.